src/test/util.ts

```diff
@@ -68,27 +68,30 @@ export function createFileSynchronizer(baseDir?: string, debugging?: boolean) {
  * and send requests up to the server, then assert on the responses from the
  * server.
  *
- * @param baseDir If given, then the test environment will be set up with a
+ * @param fixtureDir If given, then the test environment will be set up with a
  *     copy of the given basedir in os.tempdir() as the client's workspace.
  *     If this param is not passed, an empty tempdir will be used.
  * @param debugging If true, does extra logging about the requests and responses
  *     between client and server.
  */
-export async function createTestEnvironment(
-    baseDir?: string, debugging?: boolean) {
-  if (baseDir) {
-    baseDir = getTempCopy(baseDir);
-  } else {
-    baseDir = fs.mkdtempSync(path.join(os.tmpdir(), 'editor-service-tests'));
-  }
-
-  const {serverConnection, clientConnection} = createTestConnections(debugging);
+export async function createTestEnvironment(options: {
+  fixtureDir?: string,
+  debugging?: boolean,
+  capabilities?: ClientCapabilities
+} = {}) {
+  const capabilities = options.capabilities || defaultClientCapabilities;
+  const baseDir = options.fixtureDir === undefined ?
+      fs.mkdtempSync(path.join(os.tmpdir(), 'editor-service-tests')) :
```

Comment:
await mkdtemp with promisify or fsextra?

Reply:
Tests are sequential and sync methods are faster. Nothing to be gained.